orangewidget/gui.py

```diff
@@ -1726,19 +1764,29 @@ def do_commit():
         if orientation is None:
             orientation = Qt.Vertical if checkbox_label else Qt.Horizontal
         b = widgetBox(widget, box=box, orientation=orientation,
-                      addToLayout=False)
-        b.setSizePolicy(QSizePolicy.Preferred, QSizePolicy.Maximum)
+                      addToLayout=False, margin=0, spacing=0)
+        if not widget.layout().isEmpty() \
+                and _is_horizontal(orientation) \
+                and isinstance(widget.layout(), QtWidgets.QHBoxLayout):
+            separator(b, 16, 0)
 
     b.checkbox = cb = checkBox(b, master, value, checkbox_label,
                                callback=checkbox_toggled, tooltip=auto_label)
     if _is_horizontal(orientation):
-        b.layout().addSpacing(10)
+        w = b.style().pixelMetric(QStyle.PM_CheckBoxLabelSpacing)
+        separator(b, w, 0)
     cb.setSizePolicy(QSizePolicy.Preferred, QSizePolicy.Preferred)
 
     b.button = btn = VariableTextPushButton(
         b, text=label, textChoiceList=[label, auto_label], clicked=do_commit)
+    btn.setMaximumHeight(btn.sizeHint().height())
```

Comment:
Can you try setting `QSizePolicy.Maximum` vertical size policy here instead? the button's size hint can change at runtime (app font/style change, ...).